src/legacy/ui/public/courier/search_source/normalize_sort_request.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexPattern } from '../../../../core_plugins/data/public';
+import { EsQuerySortValue, SortOptions } from './types';
+
+export function normalizeSortRequest(
+  sortObject: EsQuerySortValue | EsQuerySortValue[],
+  indexPattern: IndexPattern | string | undefined,
+  defaultSortOptions: SortOptions = {}
+) {
+  const sortArray: EsQuerySortValue[] = Array.isArray(sortObject) ? sortObject : [sortObject];
+  return sortArray.map(function(sortable) {
+    return normalize(sortable, indexPattern, defaultSortOptions);
+  });
+}
+
+/**
+ * Normalize the sort description to the more verbose format (e.g. { someField: "desc" } into
+ * { someField: { "order": "desc"}}), and convert sorts on scripted fields into the proper script
+ * for Elasticsearch. Mix in the default options according to the advanced settings.
+ */
+function normalize(
+  sortable: EsQuerySortValue,
+  indexPattern: IndexPattern | string | undefined,
+  defaultSortOptions: any
```

Comment:
No, `defaultSortOptions` comes from an advanced UI setting. We could probably type it but it really should come from Elasticsearch since it represents additional parameters you can send in your `sort` portion of a search request.